api/src/controllers/UserController.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+require_once 'AppController.php';
+require_once __DIR__.'/../models/User.php';
+require_once __DIR__.'/../repository/UserRepository.php';
+
+class UserController extends AppController
+{
+    private $userRepository;
+
+    const MAX_FILE_SIZE = 2048*2048;
+    const SUPPORTED_TYPES = ['image/png', 'image/jpeg'];
+    const UPLOAD_DIRECTORY = '/../public/img/profiles/';
+
+    public function __construct() {
+        parent::__construct();
+        $this->userRepository = new UserRepository();
+    }
+
+    public function getUserDetails()
+    {
+        header('Access-Control-Allow-Origin: *');
+        header('Access-Control-Allow-Headers: Content-Type');
+        header('Content-type: application/json');
+
+        $postData = file_get_contents("php://input");
+
+        if(!$this->isGet() || empty($postData)) {
+            http_response_code(400);
+            echo json_encode("Bad request");
+            die();
+        }
+
+        $request = json_decode($postData);
+        $email = $request->email;
+
+        $user = $this->userRepository->getUser($email);
+        if(!$user) {
+            http_response_code(404);
+            echo json_encode("User not found");
+        } else {
+            http_response_code(200);
+            echo json_encode($user->toJson());
+        }
+    }
+
+    public function setUserDetails()
+    {
+        header('Access-Control-Allow-Origin: *');
+        header('Access-Control-Allow-Headers: Content-Type');
+        header('Content-type: application/json');
+
+        $postData = file_get_contents("php://input");
+        if(!$this->isPost() || empty($postData)) {
+            http_response_code(400);
+            echo json_encode("Bad request");
+            die();
+        }
+
+        $request = json_decode($postData);
+        $name = trim($request->name);
+        $surname = trim($request->surname);
+        $email = trim($request->email);
+        $password = $request->password;
+        $newPassword = str_replace(' ', '', $request->new_password);
+
+        if (empty($name) || empty($surname) || empty($email) || empty($password)) {
+            http_response_code(400);
+            echo json_encode("Bad request");
+            die();
+        }
+
+        $user = $this->userRepository->getUser($email);
+        if(!$user || !password_verify($password, $user->getPassword())) {
+            http_response_code(400);
+            echo json_encode("Wrong email or password");
+            die();
+        }
+
+        if (!ctype_alpha($name) || !ctype_alpha($surname)) {
+            http_response_code(400);
+            echo json_encode("Wrong name or surname");
+            die();
+        }
+
+        $this->validatePassword($password);
```

Comment:
Hasło, które jest generowane po requeście resetowania hasła ma 6 znaków. Sprawdzanie długości aktualnego hasła uniemożliwia nam zmianę czegokolwiek, ale imo wydaje się być trochę zbędne skoro i tak mamy ograniczenie na nowe hasło. Zawsze nam zwraca "Password too short".

![image](https://user-images.githubusercontent.com/77665045/163685032-99bf4f1b-f17a-42e7-8f79-07269f920582.png)

Reply:
racja, coś musiałem pomieszać, bo w ogóle nie chciałem walidować aktualnego hasła, tylko to nowe xD.